### PR TITLE
implement steam's rich presence.

### PR DESCRIPTION
**Pull request by surepy**
This PR aims to implement Steam's [Enhanced Rich Presence](https://partner.steamgames.com/doc/features/enhancedrichpresence) onto Quaver. 

which adds the benefit of friend grouping if in the same multiplayer lobby, and not having to respond "what are you playing" in steam messages. ~~not really, people message anyways.~~

~~important note: This needs separate testing as I cannot test it because I don't have access to Steamworks. lol. that's why this is a draft..~~ see below
this is the localization file that I would like to commit.

```
"lang"
{
	"english"
	{
		"tokens"
		{
			"#State"	"%state%"
			"#Details"	"%details%"
			"#Status"	"{#State}: {#Details}"
		}
	}
}
```

### CODE REVIEW

**Comment by Swan** [comment]
@surepy What exactly is preventing you from testing? You should have access to the majority of the features in the API already.

File: Quaver.Shared/Screens/Downloading/DownloadingScreen.cs [hunk 1]
```diff
@@ -853,6 +853,9 @@ private void SetRichPresence()
         {
             try
             {
+                SteamManager.SetRichPresence("State", "In the menus");
+                SteamManager.SetRichPresence("Details", "Downloading Maps");
+
                 DiscordHelper.Presence.Details = "Downloading Maps";
                 DiscordHelper.Presence.State = "In the menus";
                 DiscordHelper.Presence.LargeImageText = OnlineManager.GetRichPresenceLargeKeyText(ConfigManager.SelectedGameMode.Value);
```
Comment (Swan) [inline Quaver.Shared/Screens/Downloading/DownloadingScreen.cs:861]: It'd be better if we can change both Discord and Steam in one method, so we can avoid having to do this twice every single time.

**Review by Swan — CHANGES_REQUESTED** [review]

File: Quaver.Shared/Screens/Downloading/DownloadingScreen.cs (hunk 1 above)
Comment (surepy) [reply Quaver.Shared/Screens/Downloading/DownloadingScreen.cs:861]: fair enough, I'll do that.

**Comment by surepy** [comment]
> @surepy What exactly is preventing you from testing? You should have access to the majority of the features in the API already.

the Rich Presence Tester didn't work the day I made this apparently now it just decides to ¯\_(ツ)_/¯

image

**Comment by Swan** [comment]
@surepy I've uploaded the file you provided, but there's still errors being thrown in the tester.

image

**Comment by surepy** [comment]
lmao i forgot a `#`

**Review by Swan — APPROVED** [review]

**Pull request merged**